### PR TITLE
[Sofa.Core] Use pragma, single line namespaces and move forward declarations in fwd.h

### PR DESCRIPTION
**Pull request by damienmarchal**
Move some forward declaration in sofa/core/fwd.h
Use pragma once & one line namespace. 

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: Sofa/framework/Core/src/sofa/core/objectmodel/DataCallback.h [hunk 1]
```diff
@@ -91,11 +83,6 @@ class SOFA_CORE_API DataCallback : public DDGNode
 /// Import DataCallback from its private namespace into the public one.
 using _datacallback_::DataCallback;
 
-}
-
-}
-
-}
+} /// sofa::core::objectmodell
```
Comment (alxbilger) [inline Sofa/framework/Core/src/sofa/core/objectmodel/DataCallback.h]:
```suggestion
} /// sofa::core::objectmodel
```

**Comment by damienmarchal** [comment]
Conflict fixed & compiling;

**Review by fredroy — APPROVED** [review]

**Comment by fredroy** [comment]
[ci-build][force-full-build][with-all-tests]

**Comment by damienmarchal** [comment]
@fredroy CI passed :)

**Pull request merged**